### ISSUE TITLE
Black screen on iPhone 4 / iOS 9.2.1

### ISSUE THREAD

**Issue by peterpeterparker** [driftyco/cordova-plugin-wkwebview-engine#64]
Hey,

I know the following description gonna be not enough detailed but after spending hours trying to find the reason of my problem I'm just hoping for some ideas

I've got following app:

Tabs
--- Tab1
--- Tab2
--- Tab3
-> Go to Subpage
which contains a Slider full height which is in fact a wizard

Everything works on chrome debug, android debug, android device (nexus 5x) and ios debug.
But doesn't work on my iPhone 4 / iOS 9.2.1 (don't have any real other iphone to test).

When I start, everything goes fine, I select the Tab3 and then access my Subpage => Boom black page (but the tab bar is still displayed)

The stack tray in Xcode doesn't display any errors.

A tried during hours to delete and add pieces of my code to isolate the problem but couldn't figure out what is the problem.

Finally I remove the cordova-plugin-wkwebview-engine ... and everything is working fine :(

Like I said, don't except a pretty detailed answer but just hope that someone could maybe have a good idea?

Thx in advance for your advices

**Comment by peterpeterparker** [comment]
So I invested hours, made massive refactoring, memory optimization etc. in my code and still face that problem.

Also I noticed that the all app isn't broken. When I face the "black screen page", I could still drag the menu to open it. I click my tabs to set them as root and try again. It gonna fails again once, twice.....and then no more black screen page it works !??!!?!?

Recently I was thinking about memory leak possibilities (in my code or plugin whatever) but that recent discovery somehow amuse me

**Comment by peterpeterparker** [comment]
FCK YEAAAAH I find ways two solve my problem

First, I had a conceptual error. My wizard is based on something like 10 slides and I used EventEmitter to update my data everywhere which bring huge lag on iPhone 4. Refactored, changed the architecture of the wizard and it solved that point

Then, I still have an error, the "freaking black screen". I noticed, after trying to remove slide after slide that WkWebView couldn't just load as much slides as UIWebView could, always faced memory leak. Fortunately, the problem seems to be only there when I first load for the very first time my slider. My hack to solve that was then to load step after the step the slide. Pretty ugly code but that solve this issue.
